src/model/CollieRepository.ts

```diff
@@ -39,20 +39,19 @@ export class CollieRepository {
   }
 
   static async load(searchDir = "./"): Promise<CollieRepository> {
-    const absolutePath = path.resolve(searchDir);
-    const components = absolutePath.split(path.SEP);
+    let absolutePath = path.resolve(searchDir);
 
     // find parent directory containing the next best git repository
-    while (!(await fs.exists(path.join(...components, ".git")))) {
-      components.pop();
+    while (!(await fs.exists(path.join(absolutePath, ".git")))) {
+      absolutePath = path.dirname(absolutePath);
 
-      if (!components.length) {
+      if (absolutePath == path.dirname(absolutePath)) {
```

Comment:
d: iirc this will only hold true at the root of the filesystem (i.e. / or C:\ or sth) right? 

If that's the assumption here, it's good to add a comment, or even better give an explicit name to the condition like

```ts
const isAtRootOfFilesystem = absolutePath == path.dirname(absolutePath)
if (isAtRootOfFilesystem) {...}
```